docs/fundamentals/syslib-diagnostics/source-generator-overview.md

```diff
@@ -0,0 +1,103 @@
+| [SYSLIB1003][1003] | `InvalidLoggingMethodParameterNameTitle`                                                                     |
+| [SYSLIB1004][1004] | Logging class cannot be in a nested type                                                                      |
+| [SYSLIB1005][1005] | Could not find a required type definition                                                                    |
+| [SYSLIB1006][1006] | Multiple logging methods cannot use the same event ID within a class                                         |
+| [SYSLIB1007][1007] | Logging methods must return `void`                                                                           |
+| [SYSLIB1008][1008] | One of the arguments to a logging method must implement the `Microsoft.Extensions.Logging.ILogger` interface |
+| [SYSLIB1009][1009] | Logging methods must be `static`                                                                             |
+| [SYSLIB1010][1010] | Logging methods must be `partial`                                                                            |
+| [SYSLIB1011][1011] | Logging methods cannot be generic                                                                            |
+| [SYSLIB1012][1012] | Redundant qualifier in logging message                                                                       |
+| [SYSLIB1013][1013] | Don't include exception parameters as templates in the logging message                                       |
+| [SYSLIB1014][1014] | Logging template has no corresponding method argument                                                        |
+| [SYSLIB1015][1015] | Argument is not referenced from the logging message                                                          |
+| [SYSLIB1016][1016] | Logging methods cannot have a body                                                                           |
+| [SYSLIB1017][1017] | A LogLevel value must be supplied in the `LoggerMessage` attribute or as a parameter to the logging method   |
+| [SYSLIB1018][1018] | Don't include logger parameters as templates in the logging message                                          |
+| [SYSLIB1019][1019] | Couldn't find a field of type `Microsoft.Extensions.Logging.ILogger`                                         |
+| [SYSLIB1020][1020] | Found multiple fields of type `Microsoft.Extensions.Logging.ILogger`                                         |
+| [SYSLIB1021][1021] | Can't have the same template placeholders with different casing                                                           |
+| [SYSLIB1022][1022] | Can't have malformed format strings (for example, dangling `{`)                                |
+| [SYSLIB1023][1023] | Generating more than six arguments is not supported                                                            |
+<!-- commented out for now, as these do not exist.
```

Comment:
@layomia should we add these?

Reply:
We should add them; they are [checked in](https://github.com/dotnet/runtime/blob/6f5ab2b939ccadef92341affdd089b0f7c003280/docs/project/list-of-diagnostics.md) now. I'll make a follow up PR, thanks!